IronFrame/Utilities/FirewallManager.cs

```diff
@@ -43,6 +50,87 @@ public void OpenPort(int port, string name)
             firewallPolicy.Rules.Add(firewallRule);
         }
 
+        internal static string GetFormattedLocalUserSid(string windowsUsername)
+        {
+            var ntaccount = new NTAccount("", windowsUsername);
+            var sid = ntaccount.Translate(typeof(SecurityIdentifier)).Value;
+            return String.Format(CultureInfo.InvariantCulture, "D:(A;;CC;;;{0})", sid);
+        }
+
+        /// <summary>
+        /// Remove all firewall rules that match the given name
+        /// 
+        /// The implementation is a bit hacky. There is no way determine 
+        /// how many rules match a given name without iterating through 
+        /// the entire collection of rules. This method seems cheaper 
+        /// but uses exceptions to signal that no more rules match the given
+        /// name.
+        /// </summary>
+        /// <param name="userName"></param>
+        public void RemoveAllFirewallRules(string userName)
+        {
+            var firewallPolicy = (INetFwPolicy2)Activator.CreateInstance(Type.GetTypeFromProgID(NetFwPolicy2ProgID));
```

Comment:
It looks like we already have a helper in this class for getting an instance of a COM type `getComObject()`.  See existing firewall methods for usage.